lasagne/layers/helper.py

```diff
@@ -183,6 +189,13 @@ def get_output(layer_or_layers, inputs=None, **kwargs):
                                  "mapping this layer to an input expression."
                                  % layer)
             all_outputs[layer] = layer.get_output_for(layer_inputs, **kwargs)
+            if not isinstance(layer, Mock) and \
+                    not isinstance(layer.get_output_for, Mock):
+                unused_kwargs -= set(getargspec(layer.get_output_for)[0])
+            unused_kwargs -= set(layer.get_output_kwargs)
+    if unused_kwargs:
+        template = ", ".join(unused_kwargs)
+        warn("get_output() was called with unused kwargs: %s" % template)
```

Comment:
Could we go the other way instead -- collect a list of all accepted keywords and compare it to `set(kwargs.keys())` afterwards? This would allow us to not only print the keyword arguments that were not understood, but also the list of keyword arguments that would have been accepted (or even make suggestions using `difflib.get_close_matches`).